client/src/components/UIElements/SubjectOption.tsx

```diff
@@ -0,0 +1,56 @@
+import { useState } from "react";
+import "./SubjectOption.scss";
+import { subjectType } from "../../types/subjectType";
+
+interface SubjectOptionProps {
+  name: string;
+  parentId?: string;
+  containMoreSubjects?: boolean;
+  subjects?: subjectType[];
+}
+
+const SubjectOption: React.FC<SubjectOptionProps> = ({
+  name,
+  subjects,
+}) => {
+  const [isHidden, setIsHidden] = useState(true);
```

Comment:
this is abit of an anti-pattern to have a boolean that represent a negative value.
This might cause confusion, its better to have a boolean that represent a positive value.
https://www.serendipidata.com/posts/naming-guidelines-for-boolean-variables


```suggestion
  const [showChildren, setShowChildren] = useState(false);
```